src/Core/Managed/Shared/DataContracts/RequestTelemetry.cs

```diff
@@ -223,19 +223,31 @@ void ITelemetry.Sanitize()
             // Required field
             if (!this.Success.HasValue)
             {
-                int responseCode;
-
-                if (int.TryParse(this.ResponseCode, NumberStyles.Any, CultureInfo.InvariantCulture, out responseCode))
-                {
-                    this.Success = (responseCode < 400) || (responseCode == 401);
-                }
-                else
-                {
-                    this.Success = true;
-                }
+                this.Success = this.GetSuccessFromResponseCode();
             }
 
             this.Context.SanitizeTelemetryContext();
         }
+
+        internal bool GetSuccessFromResponseCode()
```

Comment:
I don't think this logic belong here. If customer override the response code - customer should also override the `Success` value. 

I'd say - remove this logic completely with the mention in CHANGELOG.md